SDWebImage/NSData+ImageContentType.h

```diff
@@ -10,15 +10,18 @@
 #import <Foundation/Foundation.h>
 #import "SDWebImageCompat.h"
 
-typedef NS_ENUM(NSInteger, SDImageFormat) {
-    SDImageFormatUndefined = -1,
-    SDImageFormatJPEG = 0,
-    SDImageFormatPNG,
-    SDImageFormatGIF,
-    SDImageFormatTIFF,
-    SDImageFormatWebP,
-    SDImageFormatHEIC
-};
+/**
+ You can use switch case like normal enum. It's also recommended to add a default case.
+ For custom coder plugin, it may also extern the enum for supported format.
+ */
+typedef NSInteger SDImageFormat NS_TYPED_EXTENSIBLE_ENUM;
+static const SDImageFormat SDImageFormatUndefined = -1;
+static const SDImageFormat SDImageFormatJPEG = 0;
+static const SDImageFormat SDImageFormatPNG = 1;
+static const SDImageFormat SDImageFormatGIF = 2;
+static const SDImageFormat SDImageFormatTIFF = 3;
+static const SDImageFormat SDImageFormatWebP = 4;
+static const SDImageFormat SDImageFormatHEIC = 5;
```

Comment:
This break further APIs. Our user and related framework always use enum and swtich case to check. 

+ `sd_imageFormatForImageData:`
+ `sd_UTTypeFromImageFormat:`
+ `sd_imageFormatFromUTType:`

If we want to change to string, why not just use Apple's [UTType](https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/understanding_utis/understand_utis_intro/understand_utis_intro.html). They are all string...

Reply:
But anyway. Using a `NS_STRING_ENUM` can solve the problem 2 : `Conflict define for same value`. Because if using string, we can do anything, like a prefix domain or something to avoid this. Instead of that wiki page to list raw value.

However, I guess this need to change branch of API's implementation. I'll try to replace with string enum and check the effected result to decide whether it's acceptable. 😴

Reply:
Changing those values to strings seems too much work for very little benefit. 
I understand @dreampiggy 's point about the lack of extensibility for `SDImageFormat`. 
I agree with `typedef NSInteger SDImageFormat NS_TYPED_EXTENSIBLE_ENUM`.
More to that, I would just add the rest we already know and support through plugins
```objective-c
static const SDImageFormat SDImageFormatAPNG = SDImageFormatPNG;
static const SDImageFormat SDImageFormatHEIF = 6;
static const SDImageFormat SDImageFormatBPG = 7;
...
```

Reply:
Yeah, we already stepped here, if it needs much works, keep the `integer` is acceptable.

Small suggestion, can we make some indentations when initialize? just like below, it looks more comfortable IMO. 😂  
![image](https://user-images.githubusercontent.com/5061845/43112953-e4e6766a-8f2a-11e8-880e-4c076731e128.png)

Reply:
@zhongwuzw :)...Does this indent looks more clear ? I don't find anything different...😂

The user who use SDWebImage, should not assume anything about the raw value. Only the developer who need provide new image foramt, should avoid duplicated raw value. So I guess maybe write something like this does not do anything....

And the original enum define also does not contains any indent as well...

Reply:
@dreampiggy It's the same, just look more beautiful IMO. 😂 

<img width="655" alt="image" src="https://user-images.githubusercontent.com/5061845/43117770-2ae3ad48-8f41-11e8-9679-bc3de13d0654.png">

Reply:
@bpoplauschi We don't need to list all image format, which not support with SDWebImage core repo. The coder plugin is designed to as a `plugin`. So the new image format define, should be listed in the plugin repo, but not define in the core repo...

Actually, why I fire this PR, because of that I try to upgrade `SDWebImageBPG` coder which support encoding BPG images. I try to define `SDImageFormatBPG` in that repo, however, find current enum does not allow Swift user to use in switch case. So I create this PR.

After this finished, I can update thoes plugin provided in [SDWebImage Organization](https://github.com/SDWebImage/). And you can see, after change this, when people import both of core lib and plugin, then can use switch case in both Objective-C and Swift. So it's no need to define this value in the core lib.

Reply:
@zhongwuzw See the new commit...